kuberos.go

```diff
@@ -358,9 +365,28 @@ func populateUser(cfg *api.Config, p *extractor.OIDCAuthenticationParams) api.Co
 			},
 		},
 	}
+
 	for name, cluster := range cfg.Clusters {
+		// If the cluster definition does not come with certificate-authority-data nor
+		// certificate-authority, then check if kuberos has access to the cluster's CA
+		// certificate and include it when possible. Assume all errors are non-fatal.
+		if len(cluster.CertificateAuthorityData) == 0 && len(cluster.CertificateAuthority) == 0 {
+			caPath := filepath.Join(DefaultAPITokenMountPath, v1.ServiceAccountRootCAKey)
+			if caFile, err := os.Open(caPath); err == nil {
+				if caCert, err := ioutil.ReadAll(caFile); err == nil {
+					cluster.CertificateAuthorityData = caCert
+				}
+			}
+		}
 		c.Clusters[name] = cluster
-		c.Contexts[name] = &api.Context{Cluster: name, AuthInfo: p.Username}
+		c.Contexts[name] = &api.Context{
+			Cluster:  name,
+			AuthInfo: p.Username,
+		}
+		// Use the first context as the current context
+		if c.CurrentContext == "" {
+			c.CurrentContext = name
+		}
```

Comment:
I'm not sure about this functionality. I personally like to avoid setting a current context, because I typically have many clusters and always want to type `kubectl --context=blah` to make sure I never accidentally operate on my current context when I actually meant to operate on another one. (i.e. I want to make it hard to set my current context to a production cluster one day, then assume it's set to the staging cluster the next day and delete a whole namespace). That said I don't use Kuberos in production at my current job, and I think I'm a little weird in this regard, so perhaps this behaviour is what most users would prefer?

Reply:
Yeah, I considered whether or not it's worth it.

We (work) recommend keeping configs for separate clusters in separate configs and setting the `KUBECONFIG` env-var to switch instead. It's obviously a different workflow, so having a default context is useful.

I'll say that this behavior does seem a little magical, though. So maybe I should just remove it, haha.